### PR TITLE
feat(insights): ActionFilter wrapping in compact modes

### LINKED ISSUE #9701: EditorPanels WIP is too cramped for Aggregated Trends steps
## Bug description

Trends steps with aggregations are too cramped to work with:

Screenshot 2022-05-09 at 19 11 40

## How to reproduce

1. With the new Editor Panels on Insights...
2. Select `Trends` and then add a step with an aggregation
3. The UI will look so cramped it is almost unusable

## Environment

- [x] PostHog Cloud (feature flagged)

Related to https://github.com/PostHog/posthog/issues/8929

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

The new sidepanel based Insight Editing is quite compact and the ActionFilters component isn't quite designed to handle this well (See https://github.com/PostHog/posthog/issues/9701)

## Changes

* Filter icon is changed to have a Bubble indicating the count rather than a lengthy word
* The ActionFilterRow is now split into 3 parts - `start`, `center`, `end` - with `center` allowing the content to wrap to a newline if needed

**Before**
Screenshot 2022-05-10 at 16 45 06

**After**
Screenshot 2022-05-10 at 16 45 37

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?
Not as much as it needs to be. I tested it on the new Insights view (currently feature-flagged) and the old one as well as checked a couple of other places. Given how crucial this component is I might add a Storybook entry so we can at least visualise all these different view modes

Storybook addition for LemonBubble

Closes #9701

### CODE REVIEW

**Comment by clarkus** [comment]
This looks like a great start to me! Wrapping the fields to multiple lines is going to be unavoidable in some cases, but I think we can reduce the likelihood of this by reducing the space required for the inline actions. With this layout we're always prioritizing actions like duplication or renaming when those are really more complimentary than essential. 

I am working on some updates that will reduce the need to wrap things, but also improve the consistency across different insight types. I should have redrawn insight query builder scenarios by the end of the day tomorrow (PST).

**Comment by benjackwhite** [comment]
Yeah I think we need to generally support wrapping in any horizontal-focused UI but I totally agree the right-hand actions are given too much weight in such a compact view. 

I'd still be tempted to get this in to make the new EditorPanels "workable" especially as they are behind a feature flag and then we can follow up with some improvements here to make it truly beautiful ;)

**Review by mariusandra — APPROVED** [review]
LGTM. @Twixes any Lemon thoughts?

File: frontend/src/lib/components/LemonBubble/LemonBubble.scss [hunk 1]
```diff
@@ -0,0 +1,93 @@
+@import '~/vars';
+
+$lemon-counter-small: 1.2rem;
+$lemon-counter-font-small: 0.6rem;
+$sf-small: 0.8;
+$sf-large: 1.2;
```
Comment (Twixes) [inline frontend/src/lib/components/LemonBubble/LemonBubble.scss:6]: What is San Francisco doing here? 😅

File: frontend/src/lib/components/LemonBubble/LemonBubble.scss [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+$position-offset: 0.5rem;
```
Comment (Twixes) [inline frontend/src/lib/components/LemonBubble/LemonBubble.scss:7]: Let's use CSS vars whenever possible (i.e. if we can avoid using Sass functions, which is almost always):
- no dependency on a preprocessor
- inheritance working just like with regular inherited CSS properties
- there's ALWAYS a single source of truth, even when working with a fully built frontend – no need to switch back and forth between the browser and editor, just edit the relevant `--variable` in devtools

File: frontend/src/lib/components/LemonBubble/LemonBubble.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    position = 'none',
+    showZero = false,
+}: LemonBubbleProps): JSX.Element {
+    console.log(showZero)
+    // NOTE: We use 1 for the text if not showing so the fade out animation looks right
+    const text = typeof count === 'number' && count !== 0 ? (count < 10 ? String(count) : '9+') : showZero ? '0' : '1'
+    const hide = count === undefined || (count == 0 && !showZero)
+
+    return (
+        <CSSTransition in={!hide} timeout={250} classNames="anim-" mountOnEnter unmountOnExit>
```
Comment (Twixes) [inline frontend/src/lib/components/LemonBubble/LemonBubble.tsx:29]: Total nits:
I've usually been using just the component name (i.e.  `classNames="LemonBubble-"`) if the `CSSTransition` applies to the whole component appearing/disappearing (then you get obvious classes like `.LemonBubble--enter-active`).
As a microinteraction 250 ms feels a _bit_ slow, maybe 200?

File: frontend/src/lib/components/LemonBubble/LemonBubble.scss [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    flex-shrink: 0;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    color: var(--white);
+    background: var(--primary);
+    border: 2px solid var(--bg-light);
+    width: $lemon-counter-small;
+    height: $lemon-counter-small;
+    font-size: $lemon-counter-font-small;
```
Comment (Twixes) [inline frontend/src/lib/components/LemonBubble/LemonBubble.scss:19]: Hmm, I'm not sure but I thought this should also have `font-weight: 600`?

File: frontend/src/lib/components/LemonBubble/LemonBubble.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+export const Sizes: ComponentStory<typeof LemonBubble> = () => {
+    return (
+        <div className="flex space-x-05">
+            <span>small:</span>
+            <LemonBubble count={4} size="small" />
+            <span>medium:</span>
+            <LemonBubble count={4} size="medium" />
+            <span>large:</span>
+            <LemonBubble count={4} size="large" />
+        </div>
```
Comment (Twixes) [inline frontend/src/lib/components/LemonBubble/LemonBubble.stories.tsx:76]: Alignment is a bit off here:
Screen Shot 2022-05-11 at 12 23 13

File: frontend/src/lib/components/icons.tsx [hunk 6]
```diff
@@ -1,17 +1,18 @@
...
-export function IconWithCount({ count, showZero, children }: PropsWithChildren<IconWithCountProps>): JSX.Element {
+export function IconWithCount({ count, children, showZero }: PropsWithChildren<IconWithCountProps>): JSX.Element {
     return (
         <span style={{ position: 'relative', display: 'inline-flex' }}>
             {children}
-            {count > 0 || showZero ? <div className="icon-count-bubble">{count < 10 ? count : '9+'}</div> : null}
+            <LemonBubble count={count} size="small" position="top-right" showZero={showZero} />
         </span>
     )
 }
```
Comment (Twixes) [inline frontend/src/lib/components/icons.tsx:18]: Nicely abstracted!

File: frontend/src/scenes/insights/ActionFilter/ActionFilter.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+export const FullWidth = Template.bind({})
+FullWidth.args = {
+    fullWidth: true,
+}
+
+export const HorizontalUI = Template.bind({})
+HorizontalUI.args = {
+    horizontalUI: true,
+}
```
Comment (Twixes) [inline frontend/src/scenes/insights/ActionFilter/ActionFilter.stories.tsx:80]: Something isn't hooked up right here: opening a filter in one story opens all of them.

File: frontend/src/scenes/insights/ActionFilter/ActionFilterRow/ActionFilterRow.tsx [hunk 8]
```diff
@@ -333,103 +339,122 @@ export function ActionFilterRow({
...
+                                flex={fullWidth ? 'auto' : undefined}
+                            >
+                                {filterElement}
+                            </Col>
+                            {customRowSuffix !== undefined && <Col className="column-row-suffix">{suffix}</Col>}
+                            {mathAvailability !== MathAvailability.None && (
+                                <>
+                                    {horizontalUI && <Col>counted by</Col>}
+                                    <Col
+                                    // style={{ maxWidth: `calc(50% - 16px${showSeriesIndicator ? ' - 32px' : ''})` }}
```
Comment (Twixes) [inline frontend/src/scenes/insights/ActionFilter/ActionFilterRow/ActionFilterRow.tsx:385]: What's this? 👀

File: frontend/src/scenes/insights/ActionFilter/ActionFilterRow/ActionFilterRow.tsx [hunk 9]
```diff
@@ -333,103 +339,122 @@ export function ActionFilterRow({
...
+                                            mathAvailability={mathAvailability}
+                                        />
+                                    </Col>
+                                    {mathDefinitions[math || '']?.onProperty && (
+                                        <>
+                                            {horizontalUI && <Col>on property</Col>}
+                                            <Col
+                                                style={
+                                                    {
+                                                        // maxWidth: `calc(50% - 16px${showSeriesIndicator ? ' - 32px' : ''})`,
```
Comment (Twixes) [inline frontend/src/scenes/insights/ActionFilter/ActionFilterRow/ActionFilterRow.tsx:402]: 👀

**Review by Twixes — CHANGES_REQUESTED** [review]
Looks great overall! I left some comments around details (mostly it's just me being a proponent of CSS vars…)

File: frontend/src/lib/components/LemonBubble/LemonBubble.scss (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/lib/components/LemonBubble/LemonBubble.scss:6]: that shorthand for `scale-factor` might have been too short ;)

File: frontend/src/lib/components/LemonBubble/LemonBubble.scss (hunk 4 above)
Comment (benjackwhite) [reply frontend/src/lib/components/LemonBubble/LemonBubble.scss]: Figma says 700 so changed it to match that 👍

File: frontend/src/scenes/insights/ActionFilter/ActionFilterRow/ActionFilterRow.tsx (hunk 8 above)
Comment (benjackwhite) [reply frontend/src/scenes/insights/ActionFilter/ActionFilterRow/ActionFilterRow.tsx:385]: Left over from me comparing what it used to be and what it is now. Will delete

File: frontend/src/scenes/insights/ActionFilter/ActionFilter.stories.tsx (hunk 7 above)
Comment (benjackwhite) [reply frontend/src/scenes/insights/ActionFilter/ActionFilter.stories.tsx:82]: I'm not certain whats going on here either. It seems to be something about the Kea logic being shared perhaps. @mariusandra do you have some pointers here? My gut tells me that the components should be wrapped in some sort of Bind to keep them separate?

File: frontend/src/lib/components/LemonBubble/LemonBubble.scss [hunk 10]
```diff
@@ -1,93 +1,94 @@
...
 
-$lemon-counter-small: 1.2rem;
-$lemon-counter-font-small: 0.6rem;
-$sf-small: 0.8;
-$sf-large: 1.2;
-$position-offset: 0.5rem;
-
 .LemonBubble {
+    --lemon-counter-small: 1.2rem;
+    --lemon-counter-font-small: 0.6rem;
```
Comment (Twixes) [inline frontend/src/lib/components/LemonBubble/LemonBubble.scss:5]: One more thing: I'm a bit confused why this is `-small` if `--scale-small` is a separate multiplier

Comment (benjackwhite) [reply frontend/src/lib/components/LemonBubble/LemonBubble.scss]: Good catch. Should be "size"

**Review by Twixes — APPROVED** [review]
Looks great

**Pull request merged**